Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseOpMode.java

```diff
@@ -127,37 +157,97 @@ public void driveWithIMU(double xPower, double yPower, double tPower) {
         }
 
         // calculate speed and direction of each individual motor and set power of motors to speed
-        motorFL.setPower(-yPower + xPower + tPower);
-        motorFR.setPower(-yPower - xPower - tPower);
-        motorBL.setPower(-yPower - xPower + tPower);
-        motorBR.setPower(-yPower + xPower - tPower);
+        motorFL.setPower(yPower + xPower + tPower);
+        motorFR.setPower(yPower - xPower - tPower);
+        motorBL.setPower(yPower - xPower + tPower);
+        motorBR.setPower(yPower + xPower - tPower);
     }
 
     /**
      * this method will allow the grabber to open or close given a boolean input
-     * @param isOpen true for open grabber and false for closed grabber
+     * @param position the position of the grabber
      */
-    public void driveGrabber(boolean isOpen) {
-        if (isOpen) {
-            servoGrabber.setPosition(Constants.GRABBER_OPEN_POSITION);
-        } else {
-            servoGrabber.setPosition(Constants.GRABBER_CLOSE_POSITION);
-        }
+    public void driveGrabber(double position) {
+        servoGrabber.setPosition(position);
     }
 
     /**
-     * this method will allow the slides to move upwards and downwards given a specified target position
-     * @param targetPosition target position for vertical slides motors in ticks
+     * this method will allow the slides to move to a specified target position
+     * @param targetPosition target position for slides motors in ticks
      */
     public void driveSlides(int targetPosition) {
+        int error = targetPosition - motorLeftSlides.getCurrentPosition();
+        double motorPower = error * Constants.SLIDE_MOTOR_KP;
+
+        // slides not yet at target position
+        if (Math.abs(error) > Constants.ROBOT_SLIDE_TOLERANCE_TICKS) {
+            // slides going down - joystick
+            if (error < 0 && error > -200) {
+                motorLeftSlides.setPower(-0.3);
+                motorRightSlides.setPower(-0.3);
+                // slides going down - bumpers
+            } else if (error < -200) {
+                motorLeftSlides.setPower(-1.0);
+                motorRightSlides.setPower(-1.0);
+                // slides going up - proportional control
+            } else {
+                motorLeftSlides.setPower(motorPower);
+                motorRightSlides.setPower(motorPower);
+            }
+            // slides at target position
+        } else {
+            motorLeftSlides.setPower(Constants.SLIDE_FEEDFORWARD);
+            motorRightSlides.setPower(Constants.SLIDE_FEEDFORWARD);
+        }
+    }
+
+    // centers the cone on the junction top
```

Comment:
All methods should have a javadoc comment

Reply:
Made javadoc comment, increased detail